sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -28,6 +28,86 @@ public class TableClient
         private readonly string _version;
         private readonly bool _isPremiumEndpoint;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TableClient"/>.
+        /// </summary>
+        /// <param name="tableName">The name of the table with which this client instance will interact.</param>
+        /// <param name="endpoint">
+        /// A <see cref="Uri"/> referencing the table service.
+        /// This is likely to be similar to "https://{account_name}.table.core.windows.net/" or "https://{account_name}.table.cosmos.azure.com/".
+        /// </param>
+        /// <param name="options">
+        /// Optional client options that define the transport pipeline policies for authentication, retries, etc., that are applied to every request.
+        /// </param>
+        public TableClient(string tableName, Uri endpoint, TableClientOptions options = null)
+            : this(tableName, endpoint, default(TableSharedKeyPipelinePolicy), options)
+        {
+            Argument.AssertNotNull(tableName, nameof(tableName));
+
+            if (endpoint.Scheme != "https")
+            {
+                throw new ArgumentException("Cannot use TokenCredential without HTTPS.", nameof(endpoint));
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TableClient"/>.
+        /// </summary>
+        /// <param name="tableName">The name of the table with which this client instance will interact.</param>
+        /// <param name="endpoint">
+        /// A <see cref="Uri"/> referencing the table service.
+        /// This is likely to be similar to "https://{account_name}.table.core.windows.net/" or "https://{account_name}.table.cosmos.azure.com/".
+        /// </param>
+        /// <param name="credential">The shared key credential used to sign requests.</param>
+        public TableClient(string tableName, Uri endpoint, TableSharedKeyCredential credential)
+            : this(tableName, endpoint, new TableSharedKeyPipelinePolicy(credential), null)
+        {
+            Argument.AssertNotNull(tableName, nameof(tableName));
```

Comment:
Should we always have an xml comment for thrown exceptions? Not sure what the policy is on that.

Reply:
yes, good catch!